services/api/src/routes/users.js

```diff
@@ -88,26 +99,82 @@ router
     };
   })
   .use(requirePermissions({ endpoint: 'users', permission: 'write', scope: 'global' }))
+  .post('/:userId/re-invite', async (ctx) => {
+    const { user, authUser } = ctx.state;
+    if (user.status !== 'invited') {
+      ctx.throw(400, `You can't invite a user that has the status ${user.status}`);
+    }
+    await sendInvite(authUser, user);
+    ctx.status = 204;
+  })
   .post(
     '/',
     validateBody(
       User.getCreateValidation({
-        password: passwordField.required(),
+        password: passwordField,
       })
     ),
     async (ctx) => {
-      const { email } = ctx.request.body;
+      const { email, password } = ctx.request.body;
       const existingUser = await User.findOne({ email });
       if (existingUser) {
         ctx.throw(400, 'A user with that email already exists');
       }
-      const user = await User.create(ctx.request.body);
+      const userFields = ctx.request.body;
+      userFields.status = password ? 'activated' : 'invited';
+
+      const user = await User.create(userFields);
+      if (!password) {
+        await sendInvite(ctx.status.authUser, user);
+      }
 
       ctx.body = {
         data: user,
       };
     }
   )
+  .post(
+    '/invite',
+    validateBody({
+      emails: Joi.array().items(Joi.string().email()).required(),
+      role: Joi.string()
+        .valid(...Object.keys(roles))
+        .required(),
+    }),
+    async (ctx) => {
+      const { authUser } = ctx.state;
+      const { emails, role } = ctx.request.body;
+
+      let userRoles = [];
+      if (role) {
+        const def = roles[role];
+        if (def.allowScopes.includes('global')) {
+          userRoles = [{ role, scope: 'global' }];
+        } else {
+          // TODO: allow for multiple scopes, leaving it up to the implementor to decide
+          // roles.roles = [{ role, scope: 'organization', scopeRef: organization.id }];
+        }
+      }
+
+      for (let email of [...new Set(emails)]) {
```

Comment:
is this just to uniqify the emails in case the same one is passed in twice? Is that a common thing?

Reply:
yeah, its just to deduplicate